crates/goose/src/agents/tool_execution.rs

```diff
@@ -54,15 +54,27 @@ impl Agent {
         permission_manager: &'a mut PermissionManager,
         message_tool_response: Arc<Mutex<Message>>,
         cancellation_token: Option<CancellationToken>,
+        inspection_results: &'a [crate::tool_inspection::InspectionResult],
     ) -> BoxStream<'a, anyhow::Result<Message>> {
         try_stream! {
-            for request in tool_requests {
+            for (_i, request) in tool_requests.iter().enumerate() {
```

Comment:
hmm, that seems wrong, _i and then iterate? I would suggest turning inspection_results into a dict w request_id as key, so you can look it up directly.

also, below you handle the case where you can't find and then set the messagae to Non, where before it had a value. should we at this point always have a value? I think we should always have a message and then not call it security_message, but just inspection_prompt